common/src/Common/Examples/Chess.hs

```diff
@@ -0,0 +1,274 @@
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE FlexibleInstances  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE RecursiveDo       #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE LambdaCase   #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE DataKinds             #-}
+
+module Common.Examples.Chess where
+-- This implements the game rules of chess, without any Reflex
+-- or frontend code.
```

Comment:
That should be haddock module comment. I forget the exact syntax it's a bit annoying but it works.